src/Components/CreateImageWizardV2/steps/Packages/Packages.tsx

```diff
@@ -1,545 +1,562 @@
+    </Tr>
+  );
 };
 
-const Packages = ({ getAllPackages, isSuccess }) => {
-  const { currentStep } = useContext(WizardContext);
-  const { change, getState } = useFormApi();
-  const [packagesSearchName, setPackagesSearchName] = useState(undefined);
-  const [filterChosen, setFilterChosen] = useState('');
-  const [chosenPackages, setChosenPackages] = useState({});
-  const [focus, setFocus] = useState('');
-  const selectedPackages = getState()?.values?.['selected-packages'];
-  const [availablePackages, setAvailablePackages] = useState(undefined);
-  const [selectedAvailablePackages, setSelectedAvailablePackages] = useState(
-    new Set()
+const Packages = () => {
+  const dispatch = useDispatch();
+
+  const arch = useAppSelector((state) => selectArchitecture(state));
+  const distribution = useAppSelector((state) => selectDistribution(state));
+  const customRepositories = useAppSelector((state) =>
+    selectCustomRepositories(state)
   );
-  const [selectedChosenPackages, setSelectedChosenPackages] = useState(
-    new Set()
+  const packages = useAppSelector((state) => selectPackages(state));
+
+  const [perPage, setPerPage] = useState(10);
+  const [page, setPage] = useState(1);
+  const [toggleSelected, setToggleSelected] = useState('toggle-available');
+
+  /*FOLLOW UP
+  const [toggleSourceRepos, setToggleSourceRepos] = useState(
+    'toggle-included-repos'
   );
-  const firstInputElement = useRef(null);
+  */
 
-  const oscapProfile = getState()?.values?.['oscap-profile'];
+  const [searchTerm, setSearchTerm] = useState<string>('');
+  const [
+    searchRpms,
+    { data: dataCustomPackages, isSuccess: isSuccessCustomPackages },
+  ] = useSearchRpmMutation();
 
-  const { data: customizations, isSuccess: isSuccessCustomizations } =
-    useGetOscapCustomizationsQuery(
+  const { data: dataDistroPackages, isSuccess: isSuccessDistroPackages } =
+    useGetPackagesQuery(
       {
-        distribution: getState()?.values?.['release'],
-        profile: oscapProfile,
+        distribution: distribution,
+        architecture: arch,
+        search: searchTerm,
       },
-      {
-        skip: !oscapProfile,
-      }
+      { skip: !searchTerm }
     );
-  useEffect(() => {
-    if (customizations && customizations.packages && isSuccessCustomizations) {
-      const oscapPackages = {};
-      for (const pkg of customizations.packages) {
-        oscapPackages[pkg] = { name: pkg };
-      }
-      updateState(oscapPackages);
-    }
-  }, [customizations, isSuccessCustomizations, updateState]);
 
-  // this effect only triggers on mount
-  useEffect(() => {
-    if (selectedPackages) {
-      const newChosenPackages = {};
-      for (const pkg of selectedPackages) {
-        newChosenPackages[pkg.name] = pkg;
-      }
-      setChosenPackages(newChosenPackages);
-    }
-  }, []);
+  const fetchCustomPackages = async () => {
+    await searchRpms({
+      apiSearchRpmRequest: {
+        search: searchTerm,
+        //@ts-ignore
```

Comment:
I'd prefer if we can figure out a way to handle the typing issue properly in this PR, I don't want to get into the habit of letting these into the code without a really good reason. :smile:

Reply:
Right now Typescript is complaining because urls type is `string[] | undefined`. It should only be `string[]`. 

Casting it using `as string[]` would work but that doesn't seem very safe, either. 

Do we ever expect this `flatMap()` to return `undefined`? If it is impossible (or we believe it to be impossible...) then maybe we can just check for that and throw an error if it is `undefined`?

Reply:
Fair enough, forgot to get back to this one. :see_no_evil: 

I don't think `flatMap` should be returning `undefined` here since `customRepositories` are initialized as an empty array which should get "flattened" to an empty array again if I understand it correctly.

Reply:
Updated based on the follow up comment